python/xnvme/library_loader.py

```diff
@@ -0,0 +1,60 @@
+"""
+    The xnvme.library_loader provides a single method to dynamically load the xNVMe
+    shared library. This intent is that the loader is invoked by the
+    'xnvme.ctypes_mapping' on package-initialization, such that the user can directly
+    get a library-handle like so:
+
+        from xnvme.ctypes_mapping import capi
+
+    This is useful since usage of the library handle can have multiple entry points.
+"""
+import ctypes
+import ctypes.util
+import os
+import platform
+import subprocess
+
+SHARED_EXT = {
+    "linux": "so",
+    "windows": "dll",
+    "darwin": "dylib",
+}
```

Comment:
You can probably get Python's `sysconfig` to report this

Reply:
Ohh, neat, ``SHLIB_SUFFIX`` looks like something that is useful for that purpose. Changing it to use that instead.

Reply:
``sysconfig.get_config_var("SHLIB_SUFFIX")`` seems like the thing to use?

Reply:
I haven't investigated it, but it (incorrectly?) provides `.so` on macOS
```
>>> import sysconfig
>>> sysconfig.get_config_var("SHLIB_SUFFIX")
'.so'
```

Reply:
No very useful then. Reverting back.

Reply:
Yup. reverted.